### PR TITLE
Changing version in manifests and meshconfig

### PR DESCRIPTION
**Pull request by metonymic-smokey**
Signed-off-by: metonymic-smokey <ahuja.aditi@gmail.com>

**Description**
This changes the versions for adapter images when running `edge-latest` for k8s. 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by metonymic-smokey** [comment]
Also pushed a fix for #3881

**Review by navendu-pottekkat — COMMENTED** [review]
Quick question: What does the version in the manifest change to for:

1. stable-latest in meshconfig
2. edge-latest in meshconfig
3. stable-v0.5.0 in meshconfig

?

**Comment by metonymic-smokey** [comment]
> Quick question: What does the version in the manifest change to for:
> 
> 1. stable-latest in meshconfig
> 2. edge-latest in meshconfig
> 3. stable-v0.5.0 in meshconfig
> 
> ?

1. stable-v0.5.49
2. edge-latest
3. stable-v0.5.0

**Comment by metonymic-smokey** [comment]
Also, the integration tests are timing out :(

**Comment by navendu-pottekkat** [comment]
@metonymic-smokey It should be:

1. stable-latest in meshconfig: stable-latest
2. edge-latest in meshconfig: edge-latest
3. stable-v0.5.0 in meshconfig: stable-v0.5.0

This is so that we are always subscribed to the latest image and it will pull in a new image only if it is in the `latest` tag.

**Comment by navendu-pottekkat** [comment]
Could you fix that as well in the PR?

The set version and channel in the meshconfig should also not be altered when you fetch what the latest release is from GitHub when the version is `latest`.

**Comment by metonymic-smokey** [comment]
@navendu-pottekkat added a separate case for `stable-latest`, now adapter manifests and meshconfig have images with tag `stable-latest`

**Comment by navendu-pottekkat** [comment]
@metonymic-smokey FYI: Lint error

**Comment by metonymic-smokey** [comment]
I think the meshconfig versions are not altered from what I know, so I don't think I need to change that.

**Comment by navendu-pottekkat** [comment]
> I think the meshconfig versions are not altered from what I know, so I don't think I need to change that.

Could you try setting the version to stable-latest and edge-latest and try running on Docker and Kubernetes? There might be a bug here.

**Comment by metonymic-smokey** [comment]
@navendu-pottekkat 
On k8s:
1. stable-latest works as expected, downloading from v0.5.49: 
image

2.  edge-latest also starts as expected, downloading from master: 
image

My laptop is facing some issues with docker so I'll run that when those are fixed.

**Comment by leecalcote** [comment]
Thank you for catching this:

"stable-latest in meshconfig: stable-latest"

This has been missing for a couple months. It'll be great to reinstitute channel subscriptions with this.

**Comment by metonymic-smokey** [comment]
Current status:
1. docker: `stable-latest` in meshconfig results in `stable-latest` in meshery.yaml but the version in meshconfig changes to stable-v0.5.49(this needs to be fixed)
2. k8s: channel subscriptions work as expected in meshconfig and deployment files.

File: mesheryctl/internal/cli/root/system/start.go [hunk 1]
```diff
@@ -350,9 +349,32 @@ func start() error {
 				manifestFiles := filepath.Join(utils.MesheryFolder, utils.ManifestsFolder)
 
 				// change version in meshery-deployment manifest
-				err = utils.ChangeManifestVersion(utils.MesheryDeployment, channel, version, filepath.Join(manifestFiles, utils.MesheryDeployment))
-				if err != nil {
-					return err
+				if currCtx.GetVersion() == "latest" && currCtx.GetChannel() == "stable" {
```
Comment (hexxdump) [inline mesheryctl/internal/cli/root/system/start.go:352]: How about setting a channel=stable, version=latest in this case? and rest of the code will be common?

Comment (metonymic-smokey) [reply mesheryctl/internal/cli/root/system/start.go:352]: agreed @hexxdump 
refactored in the latest commit.

**Pull request merged**